airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/up_to_date/pipeline.py

```diff
@@ -0,0 +1,322 @@
+
+    async def _run(self) -> StepResult:
+        if self.original_pyproject:
+            self.pyproject_path.write_text(self.original_pyproject)
+            self.logger.info(f"Restored pyproject.toml for {self.context.connector.technical_name}")
+        if self.original_poetry_lock:
+            self.poetry_lock_path.write_text(self.original_poetry_lock)
+            self.logger.info(f"Restored poetry.lock for {self.context.connector.technical_name}")
+
+        return StepResult(
+            step=self,
+            status=StepStatus.SUCCESS,
+        )
+
+
+class RegressionTest(Step):
+    """Run the regression test for the connector.
+    We test that:
+    - The original dependencies are installed in the new connector image.
+    - The dev dependencies are not installed in the new connector image.
+    - The connector spec command successfully.
+    """
+
+    context: ConnectorContext
+
+    title = "Run regression test"
+
+    async def _run(self, new_connector_container: dagger.Container) -> StepResult:
+        try:
+            await new_connector_container.with_exec(["spec"])
+            await new_connector_container.with_mounted_file(
+                "pyproject.toml", (await self.context.get_connector_dir(include=["pyproject.toml"])).file("pyproject.toml")
+            ).with_exec(["poetry", "run", self.context.connector.technical_name, "spec"], skip_entrypoint=True)
+        except dagger.ExecError as e:
+            return StepResult(
+                step=self,
+                status=StepStatus.FAILURE,
+                stderr=str(e),
+            )
+        return StepResult(
+            step=self,
+            status=StepStatus.SUCCESS,
+        )
+
+
+# TODO: consider https://packaging.pypa.io/en/latest/version.html#packaging.version.Version
+def compare_semver_versions(version1: str, version2: str) -> int:
+    """Compare two semver versions.
+    Return:
+    - 0 if the versions are equal.
+    - 1 if version1 is greater than version2.
+    - -1 if version1 is less than version2.
+    """
+    version1_parts = version1.split(".")
+    version2_parts = version2.split(".")
+    for i in range(3):
+        if int(version1_parts[i]) > int(version2_parts[i]):
+            return 1
+        if int(version1_parts[i]) < int(version2_parts[i]):
+            return -1
+    return 0
+
+
+def extract_poetry_show_version(poetry_show_result: str) -> str | None:
+    # Regular expression to match a version number pattern
+    version_pattern = r"\d+\.\d+\.\d+"
+
+    # Search for the pattern in the string
+    match = re.search(version_pattern, poetry_show_result)
+
+    # Extract and print the version number if a match is found
+    if match:
+        return match.group()
+    else:
+        return None
+
+
+async def get_current_cdk_version() -> str:
+    response = requests.get("https://pypi.org/pypi/airbyte-cdk/json")
+    response.raise_for_status()
+    return response.json()["info"]["version"]
```

Comment:
We already have this function [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/pipelines/helpers/connectors/cdk_helpers.py#L10). And it does not have to be async as you're not awaiting anything